src/app/src/components/RegisterForm.jsx

```diff
@@ -0,0 +1,245 @@
+import React, { Component } from 'react';
+import { arrayOf, bool, func, shape, string } from 'prop-types';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import Grid from '@material-ui/core/Grid';
+
+import AppGrid from '../containers/AppGrid';
+import ShowOnly from './ShowOnly';
+import Button from './Button';
+import RegisterFormField from './RegisterFormField';
+
+import {
+    updateSignUpFormInput,
+    submitSignUpForm,
+    resetAuthFormState,
+} from '../actions/auth';
+
+import {
+    OTHER,
+    registrationFieldsEnum,
+    registrationFormFields,
+    authLoginFormRoute,
+} from '../util/constants';
+
+import {
+    registrationFormValuesPropType,
+    registrationFormInputHandlersPropType,
+    userPropType,
+} from '../util/propTypes';
+
+import {
+    getValueFromEvent,
+    getCheckedFromEvent,
+} from '../util/util';
+
+class RegisterForm extends Component {
+    componentDidUpdate() {
+        const {
+            user,
+            history,
+        } = this.props;
+
+        return user
+            ? history.push(`/profile/${user.id}`)
+            : null;
+    }
+
+    componentWillUnmount() {
+        return this.props.clearForm();
+    }
+
+    render() {
+        const {
+            fetching,
+            error,
+            form,
+            inputUpdates,
+            submitForm,
+            sessionFetching,
+        } = this.props;
+
+        if (sessionFetching) {
+            return null;
+        }
+
+        const errorMessages = error && error.length
```

Comment:
Makes sense -- I'll move this down to into the rendering. I think in this case I used the ternary because `error.map` will only work when error is an array, but that check can be nested within the rest of it.

Reply:
Adjusted in 
a44cb4e